src/main/java/org/spongepowered/common/data/persistence/JsonDataFormat.java

```diff
@@ -144,18 +144,20 @@ private static Object read(JsonReader reader) throws IOException {
     private static Number readNumber(JsonReader reader) throws IOException {
         // Similar to https://github.com/zml2008/configurate/blob/master/configurate-gson/src/main/java/ninja/leaping/configurate/gson/GsonConfigurationLoader.java#L113
         // Not sure what's the best way to detect the type of number
-        double nextDouble = reader.nextDouble();
-        int nextInt = (int) nextDouble;
-        if (nextInt == nextDouble) {
-            return nextInt;
-        }
-
-        long nextLong = (long) nextDouble;
-        if (nextLong == nextDouble) {
-            return nextLong;
+        String str = reader.nextString();
+        try {
```

Comment:
Throwing exceptions (or allowing them to be thrown) is a rather expensive operation. Instead, maybe we could do a check to see if a decimal point exists in the string and parse a double based on that instead, though I don't know if that would be faster.

It would neaten this up slightly too.